pkgs/tools/package-management/nix/default.nix

```diff
@@ -46,67 +46,29 @@ let
     sha256 = "sha256-tI5nKU7SZgsJrxiskJ5nHZyfrWf5aZyKYExM0792N80=";
   };
 
-in lib.makeExtensible (self: {
-  nix_2_3 = (common rec {
-    version = "2.3.16";
-    src = fetchurl {
-      url = "https://nixos.org/releases/nix/nix-${version}/nix-${version}.tar.xz";
-      sha256 = "sha256-fuaBtp8FtSVJLSAsO+3Nne4ZYLuBj2JpD2xEk7fCqrw=";
+  inherit (builtins) baseNameOf dirOf listToAttrs map;
+  inherit (lib.lists) flatten;
+  majorVersions = [ ./2 ];
+  getSubDirs =
+    d: lib.mapAttrsToList
+      (k: v: lib.optional (v=="directory") (d + "/${k}"))
+      (builtins.readDir d);
+  versionDirs =
+    flatten (map getSubDirs majorVersions);
+  dirToKeyVal = d:
+    let
+      minor = baseNameOf d;
+      major = baseNameOf (dirOf d);
+    in {
+      name = "nix_${major}_${minor}";
+      value = callPackage (import d) {
+        inherit common patch-sqlite-exception boehmgc-nix_2_3;
+      };
     };
```

Comment:
We trade simplicity here to theoretical reduce the amount of future merge conflicts which is IMO not worth it.

Reply:
> We trade simplicity here to theoretical reduce the amount of future merge conflicts which is IMO not worth it.

I think it is unfair to describe the problem as "theoretical" simply because you are not affected by it.

The code above was attempting to be as general as possible; a lot of packages use this pattern (gcc, llvm, etc) and I was aiming for something that could be moved into `lib` as soon as it had a second user.  But since you prefer a shorter one-off solution I've switched to that in push 63a6de83ca1149fdc4a9d8930a311c7874315c2a.